src/System.Private.CoreLib/src/System/Threading/ThreadPool.cs

```diff
@@ -964,33 +809,34 @@ void IThreadPoolWorkItem.ExecuteWorkItem()
             }
         }
 
-        internal static ContextCallback ccb = new ContextCallback(WaitCallback_Context);
+        internal static readonly ContextCallback ccb = new ContextCallback(WaitCallback_Context);
 
         private static void WaitCallback_Context(Object state)
         {
             QueueUserWorkItemCallbackDefaultContext obj = (QueueUserWorkItemCallbackDefaultContext)state;
-            WaitCallback wc = obj.callback as WaitCallback;
+            WaitCallback wc = obj.callback;
             Debug.Assert(null != wc);
+            obj.callback = null;
             wc(obj.state);
         }
     }
 
     internal class _ThreadPoolWaitOrTimerCallback
     {
+        private WaitOrTimerCallback _waitOrTimerCallback;
+        private ExecutionContext _executionContext;
+        private Object _state;
         private static readonly ContextCallback _ccbt = new ContextCallback(WaitOrTimerCallback_Context_t);
         private static readonly ContextCallback _ccbf = new ContextCallback(WaitOrTimerCallback_Context_f);
 
-        private readonly WaitOrTimerCallback _waitOrTimerCallback;
-        private readonly ExecutionContext _executionContext;
-        private readonly Object _state;
-
-        internal _ThreadPoolWaitOrTimerCallback(WaitOrTimerCallback waitOrTimerCallback, Object state, bool flowExecutionContext)
+        internal _ThreadPoolWaitOrTimerCallback(WaitOrTimerCallback waitOrTimerCallback, Object state, bool compressStack)
```

Comment:
The previous parameter name is better (flowExecutionContext), I believe that name was purposefully used here since compressed stack is not relevant